File: ios/Kwift/com/lightningkite/kwift/observables/binding/serves.shared.swift
//Package: com.lightningkite.kwift.observables.binding
//Converted using Kwift2

import Foundation



extension MutableObservableProperty {
    public func serves(whilePresent: AnyObject, other: MutableObservableProperty<T>) -> Void {
        var suppress = false
        other.addAndRunWeak(whilePresent) { (ignored, value) in 
            if !suppress {
                suppress = true
                self.value = value
                suppress = false
            }
        }
        self.onChange.addWeak(whilePresent) { (ignored, value) in 
            if !suppress {
                suppress = true
                other.value = value
                suppress = false
            }
        }
    }
    public func serves(_ whilePresent: AnyObject, _ other: MutableObservableProperty<T>) -> Void {
        return serves(whilePresent: whilePresent, other: other)
    }
}
 

File: android/src/main/java/com/lightningkite/kwift/observables/binding/serves.shared.kt
package com.lightningkite.kwift.observables.binding

import com.lightningkite.kwift.AnyObject
import com.lightningkite.kwift.observables.MutableObservableProperty
import com.lightningkite.kwift.observables.observable
import com.lightningkite.kwift.rx.addWeak

fun <T> MutableObservableProperty<T>.serves(whilePresent: AnyObject, other: MutableObservableProperty<T>) {

    var suppress = false

    other.observable.addWeak(whilePresent, { ignored, value ->
        if (!suppress) {
            suppress = true
            this.value = value.value
            suppress = false
        }
    })

    this.onChange.addWeak(whilePresent, { ignored, value ->
        if (!suppress) {
            suppress = true
            other.value = value.value
            suppress = false
        }
    })
}
